extensions/wikibase/src/org/openrefine/wikibase/updates/MediaInfoEdit.java

```diff
@@ -49,6 +50,8 @@ public class MediaInfoEdit extends LabeledStatementEntityEdit {
 
     static final Logger logger = LoggerFactory.getLogger(MediaInfoEdit.class);
     protected final boolean overrideWikitext;
+    protected int startWait;
+    protected int maxWait;
```

Comment:
To me it feels unnatural to have those parameters be stored as fields of this class. The `MediaInfoEdit` class stores the representation of the edit as generated by the schema from project data. The "runtime" settings of how this edit is performed (credentials, maxlag parameter, editing tags, timeouts…) are stored in the `EditBatchProcessor` class, so I would rather put those new parameters there as they hold the same sort of data.

The parameters stored in `EditBatchProcessor` could (in the future) be exposed as a "editing settings" panel in the UI (currently, only maxlag is displayed, and probably too prominently). That would also let the user edit those waiting parameters in such a screen, which would feel consistent to me.

Reply:
Makes sense. I added the wait times to EditBatchProcessor. Not sure if they should also be defined there as I did now. Some variables come from constructor parameters, but these new variables are hard coded for now, so I don't know if that makes sense here.